examples/webgpu_loader_materialx.html

```diff
@@ -22,6 +22,7 @@
 				"imports": {
 					"three": "../build/three.webgpu.js",
 					"three/tsl": "../build/three.webgpu.js",
+					"three/webgpu": "../build/three.webgpu.js",
```

Comment:
As mentioned in #29404 I would favor just one import path, not two. Do you mind using `three/tsl` for now? Otherwise I would suggest we migrate from `three/tsl` to `three/webgpu` in all examples.

Reply:
I guess we should replace `three/tsl` to `three/webgpu` in all examples and completely remove `three` from the webgpu examples to align with the new build?

Reply:
Sounds good to me.

Reply:
by the way I realized we might need to keep the alias `"three": "../build/three.webgpu.js",` in the example for addons such as GLTFLoader which is used by both renderers... @Mugen87

Reply:
purely stylistic preference and to avoid confusion in the future as to what is `three` and what is `three/webgpu`, since they will both point to `"../build/three.webgpu.js"`. However, I’d suggest keeping it as is for now with both in the importmap (as I mentioned with the GLTFLoader case).

Reply:
Thanks! Sounds good to me.

Reply:
`three/tsl` is also a stylistic change? I wouldn't want to decide this without @mrdoob's opinion, we had a conversation about it not too long ago with @mrdoob https://github.com/mrdoob/three.js/pull/28650#issuecomment-2183949649. I'm hesitant to change something now and go back on it in a few months. Perhaps this will become clearer as we get closer to full migration?

Reply:
Which is why I suggest to merge this PR for the moment to at least fix the build on bundlers.

This PR use the following importmap:
```js
"three": "../build/three.webgpu.js",
"three/tsl": "../build/three.webgpu.js",
"three/webgpu": "../build/three.webgpu.js",
```
We can discuss about how we changes when we get closer to full migration.

Reply:
Maybe I was not too clear about the current issue here. For example if a project currently use `InstancedPointsNodeMaterial` the bundler will break as it cannot resolve anymore `InstancedPointsNodeMaterial` in `three`:
```js
import {
	Mesh,
	InstancedPointsNodeMaterial
} from 'three';
```
The new build implies that we should do this instead:
```js
import { Mesh } from 'three';

import { InstancedPointsNodeMaterial } from 'three/webgpu';
```

Reply:
For consistency, I think this PR should use `three/tsl` instead of `three/webgpu`. In other addons like the post processing nodes, we also import classes like `NodeMaterial` from `three/tsl`.

Reply:
Noted! Updated the PR consequently. The changes became very minimal which is indeed a good thing and simplify the import map of the examples.

Reply:
> In other addons like the post processing nodes, we also import classes like `NodeMaterial` from `three/tsl`.

Does that sound okay though? I feel like we should be using `three/tsl` for TSL only and not for materials.

Reply:
What was the original motivation to have `three/tsl` next to `three/webgpu`?

Maybe I'm too overcautious here but I fear having `three/tsl`, `three/webgpu` and `three` as distinct imports could make things a bit confusing for users (in the sense of "what import do I use for what module"). Regarding usability, isn't having less imports as possible better? 

My current preference would be using `three/webgpu` instead of `three/tsl` and removing `three/tsl`. Importing the TSL modules over `three/webgpu` also implies you can only using TSL in context of `WebGPURenderer`, not with `WebGLRenderer`.

Reply:
Perhaps we can keep the Node System in the core and move TSL out. I just want to ensure we are aligned on this because we’ve been planning to move the Node System to the core for nearly a decade. Let me clarify the reasoning behind this approach:

When I started implementing NodeMaterial for WebGPU, it initially relied on ShaderChunk, which made NodeMaterial dependent on three.module, similar to how things worked with WebGLRenderer. However, this approach proved limiting. For all the features the Node System offers, it required a new level of abstraction in the renderer. Since WebGPURenderer already uses the Node System in its core, we don’t need to worry about intermediating these resources, simplifying many processes.

For example, with the current lighting system, integrating TiledLighting into the base only required a few lines changes, thanks to the Node abstraction. The same applies to MRT and could extend to other systems in the future. Without Nodes, these implementations would have been far more complex.

The Node System also enables native shader code management, such as with wgslFn, and has room for improvement to better support native shader languages. Additionally, we’ve enhanced the BindingGroups system to optimize updates—differentiating between buffers updated per frame and per render.

If users had written raw shaders (non-Node System, such as raw WGSL or GLSL), they’d need to manually manage updates to bindings, group indexes, and other adjustments whenever system changes occurred. With the Node System, this overhead is eliminated, allowing us to implement significant updates without impacting existing code. The GLSL/WGSL implementations benefit from this abstraction managed by nodes, which addresses longstanding issues present in WebGLRenderer.

Now to separate the TSL from the core we will need at least:
- Convert TSL code to Nodes wherever it is used in the `core`.  
- Address the fact that many existing Nodes rely on TSL, meaning they would need to be converted to Nodes. Materials shaders, in particular, pose a challenge due to their number and complexity.  
- Rework PMREM to rely on a different abstraction, removing its dependency on TSL.  
- Update the Background system and others to use Nodes instead of TSL.  

There might be additional considerations I’m overlooking, but this seems to be the path forward if we aim to separate it as an extension. That said, this process doesn’t seem straightforward, and the benefits appear limited. The primary gains would be a cleaner import structure and potentially a slight reduction in library size. However, this is debatable, as replacing TSL with Nodes in Materials could lead to a larger codebase overall.

I just wanted to think if we have other possibilities.

Reply:
If the problem is just an alias, this should solve it:
https://shorturl.at/Q595z

```js
import * as THREE from './three.js';
import { vec3 } from './three.tsl.js';

console.log( vec3(), THREE.TSL.vec3() );
```

Reply:
After this code split PR we have this:

```js
./build/three.core.js
./build/three.webgpu.js // this file imports ./three.core.js
```

If we moved the TSL language/functions to its own file we would end with this:

```js
./build/three.core.js
./build/three.tsl.js
./build/three.webgpu.js // this file imports ./three.core.js and ./three.tsl.js
```

For that we may need a new `./src/Three.TSL.js`?

`WebGPURenderer` and `NodeMaterial` could import things from `./src/Three.TSL.js` so we don't need to convert TSL to Nodes.

But I think this would allow this usage:

```js
import * as THREE from './three.webgpu.js';
import { vec3 } from './three.tsl.js';
```

I'm just not enjoying how we are polluting the `THREE` "namespace" with all the TSL functions.

Having this separation could also inspire/motivate other people/engines to look for ways of adopt TSL too.

Reply:
> But I think you said that this approach was not tree-shakeable?

So I ended up figuring out another way to do this once we have a new file.

If we have a `three.tsl.js` file like you suggested but for the alias this will work with three-shaking. 
I posted a link to the working example https://github.com/mrdoob/three.js/pull/29644#discussion_r1852826825

Instead of `const { vec2, vec3 } = TSL;`  we would use `import { vec3 } from './three.tsl.js';`

The alias file would look like this:
```js
import { TSL } from './three.webgpu.js';  

const vec2 = TSL.vec2;
const vec3 = TSL.vec3;
// ...

export { vec3, vec2 };
```

Reply:
Sounds like we're in alignment then.

I think this approach is the most readable:

```js
import * as THREE from './three.webgpu.js';
import { vec2, vec3 } from './three.tsl.js';
```

Should we create `./src/Three.TSL.js` or `./src/TSL.js`?

Reply:
`const { vec3 } = TSL;` works differently than `const vec3 = TSL.vec3;`

### Using `const vec2 = TSL.vec2;` - tree-shakeable
![image](https://github.com/user-attachments/assets/c5830ff2-12ce-491f-a3a1-332c3842daca)

### Using `const { vec2 } = TSL` - no tree-shakeable
![image](https://github.com/user-attachments/assets/0e1bb177-085a-49f9-bac0-b6c3c66292f9)

Reply:
That seems like a foot-gun we should not expose developers to.

Lets not export `TSL` from `three.webgpu.js` then.

Lets focus on this approach:

```js
import { vec2, vec3 } from './three.tsl.js';
```

Reply:
> Should we create ./src/Three.TSL.js or ./src/TSL.js?

Whatever you prefer, you usually choose the best names :)

Reply:
Haha! Ok, lets to `./src/TSL.js` 👍